File: 010/solve.py
#!usr/bin/python

from math import sqrt

def isprime(n):
    for x in xrange(2, int(sqrt(n)+1)):
        if n%x == 0:
            return False
    return True

def primes():
    y = 2
    while True:
        pr = y
        yield pr
        y += 1
        while isprime(y) is False:
            y += 1

result = 0
for p in primes():
    if p > 2000000:
        break
    result += p

print result

File: 010/solve.go
package main

import (
	"fmt"
	"math"
)

func isPrime(n int) bool {
	for i := 2; i < int(math.Sqrt(float64(n))+1); i++ {
		if n%i == 0 {
			return false
		}
	}

	return true
}

func main() {
	primes := make(chan int)

	go func() {
		for i := 2; ; i++ {
			if isPrime(i) {
				primes <- i
			}
		}
	}()

	prime := <-primes
	sum := 0

	for prime < 2000000 {
		sum += prime
		prime = <-primes
	}

	fmt.Println(sum)
}
